patches/@endo+ses-ava+0.2.27.patch

```diff
@@ -0,0 +1,250 @@
++ * If thunk throws, log what was thrown and then rethrow it.
++ * If thunk returns a promise, immediately return a new unresolved promise.
++ * If the first promise fulfills, silently fulfill the returned promise even if
++ * the fulfillment was an error.
++ * If the first promise rejects, log the rejection reason and then reject the
++ * returned promise with the same reason.
++ * The delayed rejection of the returned promise is an observable difference
++ * from directly calling `thunk()` but will be equivalent enough for most
++ * purposes.
++ *
++ * TODO This function is useful independent of Ava, so consider moving it
++ * somewhere and exporting it for general reuse.
++ *
++ * @param {(...unknown) => unknown} func
++ * @param {unknown[]} args
++ * @param {string} name
++ * @param {Logger=} logError
+  */
+-const logErrorFirst = (func, args, name, logger = defaultLogger) => {
++const logErrorFirst = (func, args, name, logError = console.error) => {
+   let result;
+   try {
+     result = apply(func, undefined, args);
+   } catch (err) {
+-    logger(`THROWN from ${name}:`, err);
++    logError(`THROWN from ${name}:`, err);
+     throw err;
+   }
+   if (isPromise(result)) {
+     return result.then(
+       v => v,
+       reason => {
+-        logger(`REJECTED from ${name}:`, reason);
++        logError(`REJECTED from ${name}:`, reason);
+         return result;
+       },
+     );
+@@ -47,7 +62,7 @@ const logErrorFirst = (func, args, name, logger = defaultLogger) => {
+   }
+ };
+ 
+-const testerMethodsWhitelist = [
++const overrideList = [
+   'after',
+   'afterEach',
+   'before',
+@@ -56,32 +71,35 @@ const testerMethodsWhitelist = [
+   'failing',
+   'serial',
+   'only',
+-  'skip',
+ ];
+ 
+ /**
+- * @param {TesterFunc} testerFunc
+- * @param {Logger} [logger]
+- * @returns {TesterFunc} Not yet frozen!
++ * @template {Function} T
++ * @param {T} testerFunc
++ * @param {Logger} [logError]
++ * @returns {T} Not yet frozen!
+  */
+-const wrapTester = (testerFunc, logger = defaultLogger) => {
+-  /** @type {TesterFunc} */
+-  const testerWrapper = (title, implFunc, ...otherArgs) => {
+-    /** @type {ImplFunc} */
++const augmentLogging = (testerFunc, logError = console.error) => {
++  const augmented = (title, implFunc, ...otherArgs) => {
+     const testFuncWrapper = t => {
+       harden(t);
+-      return logErrorFirst(implFunc, [t, ...otherArgs], 'ava test', logger);
++      return logErrorFirst(implFunc, [t, ...otherArgs], 'ava test', logError);
+     };
+     if (implFunc && implFunc.title) {
+       testFuncWrapper.title = implFunc.title;
+     }
+     return testerFunc(title, testFuncWrapper, ...otherArgs);
+   };
+-  return testerWrapper;
++  // re-use other properties (e.g. `.always`)
++  // https://github.com/endojs/endo/issues/647#issuecomment-809010961
++  Object.assign(augmented, testerFunc);
++  // @ts-expect-error cast
++  return augmented;
+ };
+ 
++// TODO check whether this is still necessary in Ava 4
```

Comment:
Do you plan to check before this PR is R4R?

Reply:
Nope. Ava 4 is on hold to at least after MN-1 https://github.com/Agoric/agoric-sdk/issues/4895

Reply:
thanks. Resolving.